cache/src/main/scala/org/allenai/common/cache/JsonQueryCache.scala

```diff
@@ -21,11 +21,21 @@ object JsonQueryCache {
     val port: Int = config.get[Int]("port") getOrElse Protocol.DEFAULT_PORT
     val timeoutMillis: Int = config.get[Int]("timeoutMillis") getOrElse Protocol.DEFAULT_TIMEOUT
 
+    apply[V](hostname, clientPrefix, port, timeoutMillis)
+  }
+
+  def apply[V](
+    hostname: String,
+    clientPrefix: String,
```

Comment:
Bit of a nitpick, but this should probably be the first parameter to match with the other constructor.  The grouping is conceptually `(clientPrefix, (hostname, port, timeoutMillis))`.